File: principal.py
from asignatura import Asignatura
from alumno import Alumno

class Principal:

	def __init__(self):
		ingles = Asignatura('Inglés', 3)
		mates = Asignatura('Matemáticas', 2)
		juan = Alumno('Juan Pérez', 2).matricular(ingles).matricular(mates)
		juan.set_nota(ingles, 1, 4.0).set_nota(ingles, 2, 6.0).set_nota(ingles, 3, 8.0)
		antonio = Alumno('Antonio García').matricular(mates)
		antonio.set_nota(mates, 1, 2.5)
		print('	<---TESTS--->	')
		print(f'juan.media(ingles) == {juan.media(ingles)}')
		print(f'antonio.nota(mates, 2) == {antonio.nota(mates, 2)}')
		print(f'antonio.nota(mates, 1) == {antonio.nota(mates, 1)}')
		print(f'juan.aprobada(ingles) == {juan.aprobada(ingles)}')
		input('Presiona Intro para finalizar.')

if __name__ == '__main__':
	Principal()

File: Principal.java
import java.util.Scanner;
public class Principal {
    public static Scanner in = new Scanner(System.in);
    public static void main(String[] args){
        Asignatura ingles = new Asignatura("Inglés", 3);
        Asignatura mates = new Asignatura("Matemáticas", 2);
        Alumno juan = new Alumno("Juan Pérez", 2).matricular(ingles).matricular(mates);
        juan.setNota(ingles, 1, 4.0).setNota(ingles, 2, 6.0).setNota(ingles, 3, 8.0);
        Alumno antonio = new Alumno("Antonio García", 1).matricular(mates);
        antonio.setNota(mates, 1, 2.5);
        System.out.println("	<---TESTS--->	");
		System.out.println("juan.media(ingles) == " + juan.media(ingles));
		System.out.println("antonio.nota(mates, 2) == " + antonio.nota(mates, 2));
		System.out.println("antonio.nota(mates, 1) == " + antonio.nota(mates, 1));
		System.out.println("juan.aprobada(ingles) == " + juan.aprobada(ingles));
        System.out.println("Presiona Intro para finalizar.");
		in.nextLine();
    }
}
